TS-4449 header_rewrite: Better errors and debug output

zwoop opened the pull request.

atsci commented:
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/331/ for details.

atsci commented:
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/225/ for details.

jpeach commented on plugins/header_rewrite/conditions.cc:
```diff
@@ -86,9 +86,8 @@ ConditionMethod::eval(const Resources &res)
-  bool rval = static_cast<const Matchers<std::string> *>(_matcher)->test(s);
-  TSDebug(PLUGIN_NAME, "Evaluating METHOD(): %s - rval: %d", s.c_str(), rval);
-  return rval;
+  TSDebug(PLUGIN_NAME, "Evaluating METHOD()");
```
Why did you drop the value from this debug message? The one below in ``ConditionHeader`` still has it ...

jpeach commented on plugins/header_rewrite/conditions.cc:
```diff
@@ -316,12 +314,12 @@ ConditionQuery::eval(const Resources &res)
+  TSDebug(PLUGIN_NAME, "Evaluating QUERY()");
   if (NULL == res._rri) {
-    TSDebug(PLUGIN_NAME, "QUERY requires remap initialization! Evaluating to false!");
+    TSDebug(PLUGIN_NAME, "\tQUERY requires remap initialization! Evaluating to false!");
```
This seems like a configuration problem. Maybe it should be a ``TSError``.

zwoop replied:
Maybe, but that might be a different bug to fix honestly. File a Jira? We shouldn't error out here I don't think.

jpeach commented on plugins/header_rewrite/conditions.cc:
```diff
@@ -584,16 +585,17 @@ bool
-  TSDebug(PLUGIN_NAME, "Evaluating INCOMING-PORT(): %d: rval: %d", port, rval);
-  return rval;
+
+  TSDebug(PLUGIN_NAME, "Evaluating INCOMING-PORT()");
```
Print the port?

zwoop replied:
I think it prints that as part of the other code, when doing the matching.

jpeach commented on plugins/header_rewrite/regex_helper.h:
```diff
@@ -36,11 +36,13 @@ class regexHelper
-    if (regexExtra)
+    if (regexExtra) {
       pcre_free(regexExtra);
+    }
```
FWIW, ``pcre_free`` should have the same semantics as ``free(3)`` so passing ``NULL`` is a no-op.

jpeach commented:
Looks good; I just had a couple of trivial suggestions.

👍

atsci commented:
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/239/ for details.

atsci commented:
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/345/ for details.

atsci commented:
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/240/ for details.

zwoop commented:
I cleaned up a few of the missing pieces, made a get() method better (less STL copy on use), and added some typedef's as introduced in one of the operators (it makes sense and cleans up the usage in other places).
